### PR TITLE
Add basic VMI functionality tests

### PR DESCRIPTION
**Pull request by arachmani**
Add the following tests to test kubevirt component:
- Create a VMI
- Delete a VMI

Signed-off-by: Asaf Rachmani <arachman@redhat.com>

### CODE REVIEW

**Comment by arachmani** [comment]
ci test please

**Comment by djzager** [comment]
@arachmani I think you'll need to rebase now since the MRO has been removed Then this should be g2g.

File: tests/func-tests/virtual_machines_test.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+	var vmi *kubevirtv1.VirtualMachineInstance
+	var vmiRandName = vmiName + rand.String(48)
+	vmi = kubevirtv1.NewMinimalVMIWithNS(testNamespace, vmiRandName)
+
+	Context("vmi test", func() {
+		It("should create a VirtualMachineInstance", func() {
+			Eventually(func() error {
+				_, err := client.VirtualMachineInstance(testNamespace).Create(vmi)
+				return err
+			}, timeout, pollingInterval).Should(Not(HaveOccurred()), "failed to create a vmi")
```
Comment (davidvossel) [inline tests/func-tests/virtual_machines_test.go]: do we want to make sure the VMI comes online as well here? so vmi.Status.Phase = Running?

File: tests/func-tests/virtual_machines_test.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+		})
+
+		It("should verify vmi Pashe is Running", func() {
+			Eventually(func() bool {
+				vmi, err = client.VirtualMachineInstance(testNamespace).Get(vmiRandName, &k8smetav1.GetOptions{})
```
Comment (davidvossel) [inline tests/func-tests/virtual_machines_test.go]: each individual test should be self contained and not pend on other test cases. For example, this test that waits for vmi.Status.Phase == Running is dependent on that VMI getting created in another test. Instead, this test should both create the VMI, ensure that it reaches the Running Phase, and then delete the VMI.

There only needs to be a single It() defined for this

**Comment by arachmani** [comment]
ci test please

**Comment by arachmani** [comment]
ci test please

**Comment by arachmani** [comment]
ci test please

**Comment by rthallisey** [comment]
ci add to whitelist

**Comment by rthallisey** [comment]
ci test please

File: hack/build-functest.sh [hunk 3]
```diff
@@ -0,0 +1,12 @@
+set -euo pipefail
+
+go get github.com/mattn/goveralls
+go get -v github.com/onsi/ginkgo/ginkgo
+go get -v github.com/onsi/gomega
+export PATH=$PATH:$HOME/gopath/bin
+
```
[the remark this answers is not in the document]
Comment (arachmani) [reply hack/build-functest.sh]: Yes, it's possible, but IMO it's best to stay consistent and have one file for functional-tests and a second for unit-tests (a matter of semantics)

[the remark this answers is not in the document]
Comment (rthallisey) [reply hack/build-functest.sh]: understood, but the scripts function in the same way.  I"m wondering if the overlap can be combined into a single script and called by each test suite.

File: hack/in-docker.sh [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+HCO_DIR="$(readlink -f $(dirname $0)/../)"
+BUILD_DIR=${HCO_DIR}/tests/build
+WORK_DIR="/go/src/kubevirt.io/hyperconverged-cluster-operator"
+BUILD_TAG="hco-test-build"
+
+# Build the encapsulated compile and test container
+(cd ${BUILD_DIR} && docker build --tag ${BUILD_TAG} .)
+
+# Execute the build
+[ -t 1 ] && USE_TTY="-it"
```
[the remark this answers is not in the document]
Comment (arachmani) [reply hack/in-docker.sh]: For debugging purposes, to make sure that the file descriptor points to the terminal

**Comment by rthallisey** [comment]
ci test please

**Comment by rthallisey** [comment]
looks good @arachmani .  For some reason 4.2 & the 1.15 failed to start. Let's see if re triggering helps.

File: hack/in-docker.sh (hunk 4 above)
[the remark this answers is not in the document]
Comment (arachmani) [reply hack/in-docker.sh]: I saw usage of it in CDI and kubevirt as well, so I assume that there is such a chance

**Comment by arachmani** [comment]
> looks good @arachmani . For some reason 4.2 & the 1.15 failed to start. Let's see if re triggering helps.

4.2 was updated recently and https://github.com/kubevirt/hyperconverged-cluster-operator/pull/324 should fix it

**Comment by rthallisey** [comment]
ci test please

**Pull request merged**